### ISSUE TITLE
(aws-ecs-patterns): ECS Exec Support

### ISSUE THREAD

**Issue by ghost** [aws/aws-cdk#15197]
Add ECS Exec Support to the higher level constructs in ecs-patterns

### Use Case

As a user, I want to set the flag `enableExecuteCommand: true` in my `aws-ecs-patterns.ApplicationLoadBalancedFargateService` construct.

### Proposed Solution
This is currently possible with the `ecs` module but not the `ecs-patterns` (see https://docs.aws.amazon.com/cdk/api/latest/docs/aws-ecs-readme.html#ecs-exec-command)

Relevant PRs: https://github.com/aws/aws-cdk/issues/13618
https://github.com/aws/aws-cdk/pull/14670

Solution would be to enable this flag in the higher level ecs-patterns constructs as well.

---

This is a :rocket: Feature Request

**Comment by xavierbuspatrol** [comment]
FYI, I found a workaround using Escape Hatches https://docs.aws.amazon.com/cdk/latest/guide/cfn_layer.html

My workaround is (Python) :
```        
cfn_service = self.web_service.service.node.default_child
cfn_service.add_override("Properties.EnableExecuteCommand", True)
```

**Comment by danwiltshire** [comment]
@xavierbuspatrol Did you create other resources manually such as the IAM policy for `ssmmessages`, etc?

**Comment by richardnagle** [comment]
@danwiltshire I've got this, it's C# but hopefully will help:

**Comment by mzizzi** [comment]
Working on this one in #15497 

Appologies for not giving a heads up ahead of time. I found this issue after working on a patch.

**Comment by camharris** [comment]
Here's how I handled this in typescript, maybe it's helpful for others. I added the AWS managed police for `ssmmessages`.

```typescript
        const ecsDeployment = new ecs_patterns.ApplicationLoadBalancedFargateService(this, 'EcsPattern', {
          ....
        });

        // Add need policy for EnabledExecuteCommand
        ecsDeployment.taskDefinition.taskRole.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonSSMManagedInstanceCore'))

        // Use escape hatch to add EnabledExecuteCommand to cf template
        const cfnService = ecsDeployment.service.node.defaultChild as ecs.CfnService;
        cfnService.addPropertyOverride('EnableExecuteCommand', true)
 
```

**Comment by wyjiao** [comment]
> Here's how I handled this in typescript, maybe it's helpful for others. I added the AWS managed police for `ssmmessages`.
> 
> ```ts
>         const ecsDeployment = new ecs_patterns.ApplicationLoadBalancedFargateService(this, 'EcsPattern', {
>           ....
>         });
> 
>         // Add need policy for EnabledExecuteCommand
>         ecsDeployment.taskDefinition.taskRole.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonSSMManagedInstanceCore'))
> 
>         // Use escape hatch to add EnabledExecuteCommand to cf template
>         const cfnService = ecsDeployment.service.node.defaultChild as ecs.CfnService;
>         cfnService.addPropertyOverride('EnableExecuteCommand', true)
>  
> ```

You can do "cfnService.enableExecuteCommand = true;" instead of addPropertyOverride since CloudFormation already supports the feature. This gives more type safety.

**Comment by kpeters-cbsi** [comment]
I tried this with `QueueProcessingFargateService` and got `TypeError: Cannot set property 'enableExecuteCommand' of undefined`

b0rked code:
 ```
    const service = new QueueProcessingFargateService(this, 'Service', {
        ....
    });
    // Kludge to enable execute command on this service
    const cfnService = service.node.defaultChild as CfnService
    cfnService.addPropertyOverride('EnableExecuteCommand', true) // barfs; cfnService is undefined
```
What I ended up having to do was find the `FargateService` instance and do `node.defaultChild` on that. Viz:

```
    const service = new QueueProcessingFargateService(this, 'Service', {
        ....
    });
    // Kludge to enable execute command on this service
    const fargateService = service.node.children.filter(
      (child) => child.constructor.name === 'FargateService'
    )[0] as FargateService
    const cfnService = fargateService.node.defaultChild as CfnService
    cfnService.enableExecuteCommand = true
```

**Comment by TheRealAmazonKendra** [comment]
Closed by #18663.
